zerver/data_import/slack.py

```diff
@@ -57,6 +57,37 @@
 SlackToZulipRecipientT = Dict[str, int]
 
 
+class SlackBotEmail:
+    duplicate_email_count: Dict[str, int] = {}
+    # Mapping of `bot_id` to final email assigned to the bot.
+    assigned_email: Dict[str, str] = {}
+
+    @classmethod
+    def generate_email(cls: Any, user_profile: ZerverFieldsT, domain_name: str) -> str:
+        if "real_name_normalized" in user_profile:
+            slack_bot_name = user_profile["real_name_normalized"]
+        elif "first_name" in user_profile:
+            slack_bot_name = user_profile["first_name"]
+        else:
+            raise AssertionError("Could not identify bot type")
+
+        email = slack_bot_name.replace("Bot", "").replace(" ", "") + f"-bot@{domain_name}"
+
+        if email in cls.duplicate_email_count:
+            email_prefix, email_suffix = email.split("@")
+            email_prefix += cls.duplicate_email_count[email]
+            email = "@".join([email_prefix, email_suffix])
+            # Increment the duplicate email count
+            cls.duplicate_email_count[email] += 1
+        else:
+            cls.duplicate_email_count[email] = 1
```

Comment:
I think a correct algorithm here probably needs to involve a loop or something that iterates until it finds a value not used, since it's theoretically possible that the original namespace had a `bot@example.com` and a `bot1@example.com`, and we don't want collisions in that case.

Reply:
That actually won't create an issue, as we are appending an extra `-bot` to the `slack_bot_name` while creating the email. So even if there was originally some number ahead of the `slack_bot_name`, appending of `-bot` will normalise it and there won't be any further collisions.